pymeasure/instruments/hp/hplegacyinstrument.py

```diff
@@ -0,0 +1,218 @@
+            place += 4
+        return bcd
+
+    def __str__(self):
+        """
+        Returns a pretty formatted string showing the status of the instrument
+
+        """
+        ret_str = ""
+        for field in self._fields_:
+            ret_str = ret_str + f"{field[0]}: {hex(getattr(self, field[0]))}\n"
+
+        return ret_str
+
+    def __getattribute__(self, name):
+        val = super().__getattribute__(name)
+        if name == "fields":
+            return val
+        if name in self.fields():
+            process = super().__getattribute__('_get_process_')
+            if name in process:
+                val = process[name](val)
+
+        return val
+
+    def fields(self):
+        return [desc[0] for desc in super().__getattribute__('_fields_')]
+
+
+class PackedBitsBase(ctypes.BigEndianStructure):
+    """
+    A bitfield structure containing the assignments for the status decoding
+    """
+    _pack_ = 1
+    _get_process_ = {}
+    _set_process_ = {}
+
+    def __float__(self):
+        """
+        Return a float value from the packed data of the HP3437A
+
+        """
+        # range decoding
+        # (cf table 3-2, page 3-5 of the manual, HPAK document 9018-05946)
+        # 1 indicates 0.1V range
+        if self.range == 1:
+            cur_range = 0.1
+        # 2 indicates 10V range
+        if self.range == 2:
+            cur_range = 10.0
+        # 3 indicates 1V range
+        if self.range == 3:
+            cur_range = 1.0
+
+        signbit = 1
+        if self.sign_bit == 0:
+            signbit = -1
+
+        return (
+            cur_range * signbit * (
+                self.MSD + float(self.SSD) / 10 + float(self.TSD) / 100 + float(self.LSD) / 1000
+            )
+        )
+
+    def __getattribute__(self, name):
+        val = super().__getattribute__(name)
+        if name == "fields":
+            return val
+        if name in self.fields():
+            process = super().__getattribute__('_get_process_')
+            if name in process:
+                val = process[name](val)
+
+        return val
+
+    def fields(self):
+        return [desc[0] for desc in super().__getattribute__('_fields_')]
+
+
+class HPLegacyInstrument(Instrument):
+    """
+    Class for legacy HP instruments from the era before SPCI, based on `pymeasure.Instrument`
+
+    """
+    status_desc = StatusBitsBase  # To be overriden by subclasses
+
+    def __init__(self, adapter, name, status_bitfield, **kwargs):
+        super().__init__(
+            adapter, name,
+            includeSCPI=False,
+            **kwargs,
+        )
+
+        self.status_bytes_count = ctypes.sizeof(self.status_desc)
+        self.status_bits = self.status_desc
+
+        log.info(f"Initializing {self.name}")
+
+    def write(self, command):
+        if command == "B":
+            self.adapter.connection.write("B", termination="")
+        else:
+            # super().write(command)
+            self.adapter.connection.write(command)
```

Comment:
I am not sure why `super` is commented out. To me using `super  `is a preferable alternative, but I am not sure if there is any issue here.

Reply:
changed where possible to the super call.
was probably a left over from earlier